bundles/net.tourbook/src/net/tourbook/ui/views/nutrition/DialogSearchProduct.java

```diff
@@ -0,0 +1,861 @@
+            _btnAdd.addSelectionListener(widgetSelectedAdapter(selectionEvent -> onAddProduct()));
+            setButtonLayoutData(_btnAdd);
+         }
+         {
+            /*
+             * Label: Search Type
+             */
+            _lblSearchType = UI.createLabel(headerContainer, Messages.Dialog_SearchProduct_Label_SearchType);
+            GridDataFactory.fillDefaults()
+                  .align(SWT.LEFT, SWT.CENTER)
+                  .applyTo(_lblSearchType);
+         }
+         {
+            /*
+             * combo: Search type
+             */
+            _comboSearchType = new Combo(headerContainer, SWT.READ_ONLY);
+            _comboSearchType.setVisibleItemCount(2);
+            _comboSearchType.addSelectionListener(widgetSelectedAdapter(selectionEvent -> validateFields()));
+            GridDataFactory.fillDefaults()
+                  .align(SWT.LEFT, SWT.CENTER)
+                  .grab(true, false)
+                  .applyTo(_comboSearchType);
+         }
+      }
+
+      _queryViewer = new ComboViewer(_comboSearchQuery);
+      _queryViewer.setContentProvider(new SearchContentProvider());
+      _queryViewer.setComparator(new ViewerComparator());
+   }
+
+   private void createUI_20_Viewer(final Composite parent) {
+
+      /*
+       * table viewer: products
+       */
+      final Table productsTable = new Table(parent, SWT.SINGLE | SWT.FULL_SELECTION);
+      GridDataFactory.fillDefaults().grab(true, true).applyTo(productsTable);
+      productsTable.setLinesVisible(_prefStore.getBoolean(ITourbookPreferences.VIEW_LAYOUT_DISPLAY_LINES));
+      productsTable.setHeaderVisible(true);
+
+      final GridData gdDescription = (GridData) productsTable.getLayoutData();
+      gdDescription.heightHint = _pc.convertHeightInCharsToPixels(15);
+
+      // column: Barcode
+      final TableColumn columnCode = new TableColumn(productsTable, SWT.TRAIL);
+      columnCode.setText(Messages.Tour_Nutrition_Column_Code);
+      columnCode.setWidth(_pc.convertWidthInCharsToPixels(20));
+      columnCode.addListener(SWT.Selection, _columnSortListener);
+      // column: Name
+      final TableColumn columnName = new TableColumn(productsTable, SWT.TRAIL);
+      columnName.setText(Messages.Tour_Nutrition_Column_Name);
+      columnName.setWidth(_pc.convertWidthInCharsToPixels(55));
+      columnName.addListener(SWT.Selection, _columnSortListener);
+      // column: Quantity
+      final TableColumn columnQuantity = new TableColumn(productsTable, SWT.TRAIL);
+      columnQuantity.setText(Messages.Tour_Nutrition_Column_Quantity);
+      columnQuantity.setWidth(_pc.convertWidthInCharsToPixels(15));
+      columnQuantity.addListener(SWT.Selection, _columnSortListener);
+
+      _productsViewer = new TableViewer(productsTable);
+
+      _productsViewer.setContentProvider(new ViewContentProvider());
+      _productsViewer.setComparator(_productComparator);
+      _productsViewer.setLabelProvider(new ViewLabelProvider());
```

Comment:
I found interresting bar codes

![image](https://github.com/mytourbook/mytourbook/assets/1283445/c4f08c4c-b1e8-4be7-8202-b5a0b02dc304)

![image](https://github.com/mytourbook/mytourbook/assets/1283445/dae8410c-bda7-438f-b750-9c306b5ac8c3)

Reply:
Maybe the OpenFoodFacts test data ?